src/server/pachyderm_test.go

```diff
@@ -1146,85 +1146,165 @@ func TestPipelineErrorHandling(t *testing.T) {
 	c := getPachClient(t)
 	require.NoError(t, c.DeleteAll())
 
-	dataRepo := tu.UniqueString("TestPipelineErrorHandling_data")
-	require.NoError(t, c.CreateRepo(dataRepo))
+	t.Run("ErrCmd", func(t *testing.T) {
 
-	_, err := c.PutFile(dataRepo, "master", "file1", strings.NewReader("foo\n"))
-	require.NoError(t, err)
-	_, err = c.PutFile(dataRepo, "master", "file2", strings.NewReader("bar\n"))
-	require.NoError(t, err)
-	_, err = c.PutFile(dataRepo, "master", "file3", strings.NewReader("bar\n"))
-	require.NoError(t, err)
+		dataRepo := tu.UniqueString("TestPipelineErrorHandling_data")
+		require.NoError(t, c.CreateRepo(dataRepo))
 
-	// In this pipeline, we'll have a command that fails for files 2 and 3, and an error handler that fails for file 2
-	pipeline := tu.UniqueString("pipeline1")
-	_, err = c.PpsAPIClient.CreatePipeline(
-		context.Background(),
-		&pps.CreatePipelineRequest{
-			Pipeline: client.NewPipeline(pipeline),
-			Transform: &pps.Transform{
-				Cmd:      []string{"bash"},
-				Stdin:    []string{"if", fmt.Sprintf("[ -a pfs/%v/file1 ]", dataRepo), "then", "exit 0", "fi", "exit 1"},
-				ErrCmd:   []string{"bash"},
-				ErrStdin: []string{"if", fmt.Sprintf("[ -a pfs/%v/file3 ]", dataRepo), "then", "exit 0", "fi", "exit 1"},
-			},
-			Input: client.NewPFSInput(dataRepo, "/*"),
+		_, err := c.PutFile(dataRepo, "master", "file1", strings.NewReader("foo\n"))
+		require.NoError(t, err)
+		_, err = c.PutFile(dataRepo, "master", "file2", strings.NewReader("bar\n"))
+		require.NoError(t, err)
+		_, err = c.PutFile(dataRepo, "master", "file3", strings.NewReader("bar\n"))
+		require.NoError(t, err)
+
+		// In this pipeline, we'll have a command that fails for files 2 and 3, and an error handler that fails for file 2
+		pipeline := tu.UniqueString("pipeline1")
+		_, err = c.PpsAPIClient.CreatePipeline(
+			context.Background(),
+			&pps.CreatePipelineRequest{
+				Pipeline: client.NewPipeline(pipeline),
+				Transform: &pps.Transform{
+					Cmd:      []string{"bash"},
+					Stdin:    []string{"if", fmt.Sprintf("[ -a pfs/%v/file1 ]", dataRepo), "then", "exit 0", "fi", "exit 1"},
+					ErrCmd:   []string{"bash"},
+					ErrStdin: []string{"if", fmt.Sprintf("[ -a pfs/%v/file3 ]", dataRepo), "then", "exit 0", "fi", "exit 1"},
+				},
+				Input: client.NewPFSInput(dataRepo, "/*"),
+			})
+		require.NoError(t, err)
+
+		var jobInfos []*pps.JobInfo
+		require.NoError(t, backoff.Retry(func() error {
+			jobInfos, err = c.ListJob(pipeline, nil, nil, -1, true)
+			require.NoError(t, err)
+			if len(jobInfos) != 1 {
+				return fmt.Errorf("expected 1 job, got %d", len(jobInfos))
+			}
+			return nil
+		}, backoff.NewTestingBackOff()))
+		jobInfo, err := c.PpsAPIClient.InspectJob(context.Background(), &pps.InspectJobRequest{
+			Job:        jobInfos[0].Job,
+			BlockState: true,
 		})
```

Comment:
From here to 1177 can all be replaced by a single call to `FlushJob` which should be more reliable, and more concise.